src/js/static.js

```diff
@@ -8,27 +8,52 @@
 "use strict";
 var fluid = require("infusion");
 var gpii  = fluid.registerNamespace("gpii");
-fluid.registerNamespace("gpii.express.router.static");
 
 var express = require("express");
 
-// The Express module is actually named "static", so for clarity I want to use that name in spite of the warnings in older versions of JSHint
-/* jshint -W024 */
-gpii.express.router["static"].init = function (that) {
-    fluid.each(gpii.express.expandPaths(that.options.content), function (contentDir) {
-        that.router.use(express["static"](contentDir, that.options.staticMiddlewareOptions));
-    });
+fluid.registerNamespace("gpii.express.middleware.static");
+
+gpii.express.middleware["static"].init = function (that) {
+    that.staticMiddleware = express["static"](that.options.contentDir, that.options.staticMiddlewareOptions);
+};
+
+gpii.express.middleware["static"].getMiddlewareFn = function (that) {
+    return that.staticMiddleware;
 };
 
+fluid.defaults("gpii.express.middleware.static", {
+    gradeNames: ["gpii.express.middleware"],
+    members: {
+        staticMiddleware: fluid.fail
+    },
+    invokers: {
+        "getMiddlewareFn": {
+            funcName: "gpii.express.middleware.static.getMiddlewareFn",
+            args:     ["{that}"]
+        }
+    },
+    listeners: {
+        "onCreate.init": {
+            funcName: "gpii.express.middleware.static.init",
```

Comment:
This might end up being too late (in some future usages). It seems safe enough to write
staticMiddleware: "@expand:gpii.express.middleware.static.init({that}.options.contentDir, {that}.options.staticMiddlewareOptions)"
or the less compact syntax if you prefer
(note that you will need to hoist the options directly into arguments to force evaluation)

Reply:
Fixed here and in the serveIndex wrapped middleware as well.